code/modules/mining/equipment/kinetic_crusher/kinetic_crusher.dm

```diff
@@ -0,0 +1,271 @@
+	if(!charged)
+		return
+	var/turf/proj_turf = user.loc
+	if(!isturf(proj_turf))
+		return
+	var/obj/projectile/destabilizer/destabilizer = new(proj_turf)
+	destabilizer.icon_state = "[projectile_icon]"
+	for(var/obj/item/crusher_trophy/attached_trophy as anything in trophies)
+		attached_trophy.on_projectile_fire(destabilizer, user)
+	destabilizer.aim_projectile(target, user, modifiers)
+	destabilizer.firer = user
+	destabilizer.fired_from = src
+	playsound(user, 'sound/items/weapons/plasma_cutter.ogg', 100, TRUE)
+	destabilizer.fire()
+	charged = FALSE
+	update_appearance()
+	attempt_recharge_projectile()
+
+/// Handles the timer for reloading the projectile
+/obj/item/kinetic_crusher/proc/attempt_recharge_projectile(set_recharge_time)
+	if(!set_recharge_time)
+		set_recharge_time = charge_time
+	deltimer(charge_timer)
+	charge_timer = addtimer(CALLBACK(src, PROC_REF(recharge_projectile)), set_recharge_time, TIMER_STOPPABLE)
+
+/// Recharges the projectile
+/obj/item/kinetic_crusher/proc/recharge_projectile()
+	if(!charged)
+		charged = TRUE
+		update_appearance()
+		playsound(src.loc, 'sound/items/weapons/kinetic_reload.ogg', 60, TRUE)
+
+/obj/item/kinetic_crusher/ui_action_click(mob/user, actiontype)
+	set_light_on(!light_on)
+	playsound(user, 'sound/items/weapons/empty.ogg', 100, TRUE)
+	update_appearance()
+
+/obj/item/kinetic_crusher/on_saboteur(datum/source, disrupt_duration)
+	. = ..()
+	set_light_on(FALSE)
+	playsound(src, 'sound/items/weapons/empty.ogg', 100, TRUE)
+	return TRUE
+
+/obj/item/kinetic_crusher/update_icon_state()
+	inhand_icon_state = "[current_inhand_icon_state][HAS_TRAIT(src, TRAIT_WIELDED)]" // this is not icon_state and not supported by 2hcomponent
+	return ..()
+
+/obj/item/kinetic_crusher/update_overlays()
+	. = ..()
+	if(!charged)
+		. += "[icon_state]_uncharged"
+	if(light_on)
+		. += "[icon_state]_lit"
+
+/obj/item/kinetic_crusher/compact //for admins
+	name = "compact kinetic crusher"
+	w_class = WEIGHT_CLASS_NORMAL
+
+//destablizing force
+/obj/projectile/destabilizer
+	name = "destabilizing force"
+	damage = 0 //We're just here to mark people. This is still a melee weapon.
+	damage_type = BRUTE
+	armor_flag = BOMB
+	range = 6
+	log_override = TRUE
+	/// Has this projectile been boosted
+	var/boosted = FALSE
+
+/obj/projectile/destabilizer/Initialize(mapload)
+	. = ..()
+	AddComponent(/datum/component/parriable_projectile, parry_callback = CALLBACK(src, PROC_REF(on_parry)))
+
+/obj/projectile/destabilizer/proc/on_parry(mob/user)
+	SIGNAL_HANDLER
+	boosted = TRUE
+	// Get a bit of a damage/range boost after being parried
+	damage = 10
+	range = 9
+
+/obj/projectile/destabilizer/on_hit(atom/target, blocked = 0, pierce_hit)
+	var/obj/item/kinetic_crusher/used_crusher
+	if(istype(fired_from, /obj/item/kinetic_crusher))
+		used_crusher = fired_from
+
+	if(isliving(target))
+		for(var/obj/item/crusher_trophy/crusher_trophy as anything in used_crusher.trophies)
```

Comment:
seeing as there's a chance for `used_crusher` to be null here, this would runtime